vsgenstubs/_vapoursynth.part.pyi

```diff
@@ -404,7 +406,7 @@ class Core:
     def list_functions(self) -> str: ...
 
     def register_format(self, color_family: ColorFamily, sample_type: SampleType, bits_per_sample: int, subsampling_w: int, subsampling_h: int) -> Format: ...
-    def get_format(self, id: int) -> Format: ...
+    def get_format(self, id: typing.Union[Format, int]) -> Format: ...
```

Comment:
I disagree with this. `get_format` directly calls `getFormatPreset`, which accepts and `int` and an `int` alone. When cast as an `int`, `vs.Format` happens to (intentionally) reduce to the id that `getFormatPreset` expects. In this case, it'd be more correct to explicitly call `get_format(int(my_format_object))`.

Reply:
I think there is no reason to do this: `get_format(my_format_object.id)`

Either you get a copy of the `Format`-object or you have more than one core and then the code is almost certainly a bug.

Reply:
This is about what `get_format()` accepts without erroring, and *should* accept according to the code. A `Format` object's `__int__` method allows it to be reliably accepted so IMO it fits the typehint.
```py
In [1]: import vapoursynth                                                                      

In [2]: core = vapoursynth.core                                                                 

In [3]: vapoursynth.RGB24                                                                       
Out[3]: <PresetFormat.RGB24: 2000010>

In [4]: core.get_format(vapoursynth.RGB24)                                                      
Out[4]: <vapoursynth.Format at 0x7f4051c69760>

In [5]: core.get_format(2000010)                                                                
Out[5]: <vapoursynth.Format at 0x7f4053dcd640>

In [6]: clip = core.std.BlankClip(format=vapoursynth.RGB24)                                     

In [7]: clip2 = core.std.BlankClip(format=clip.format.replace(bits_per_sample=30))              

In [8]: clip2.format                                                                            
Out[8]: <vapoursynth.Format at 0x7f40525f7e80>

In [9]: clip2.format.id
Out[9]: 2001000

In [10]: core.get_format(clip2.format)
Out[10]: <vapoursynth.Format at 0x7f4051c9c220>

In [11]: core.get_format(clip2.format.id)
Out[11]: <vapoursynth.Format at 0x7f40527b2f40>

In [12]: core.get_format(vapoursynth.RGB24) == core.get_format(2000010)
Out[12]: True

In [13]: core.get_format(vapoursynth.RGB24) is core.get_format(2000010)
Out[13]: False

In [14]: core.get_format(clip2.format) == core.get_format(clip2.format.id)
Out[14]: True

In [15]: core.get_format(clip2.format) is core.get_format(clip2.format.id)
Out[15]: False

In [16]: core.get_format(core.get_format(core.get_format(vapoursynth.RGB24)))
Out[16]: <vapoursynth.Format at 0x7f40527a5a60>

In [17]: core.get_format(core.get_format(core.get_format(2001000)))  # clip2's RGB90 format
Out[17]: <vapoursynth.Format at 0x7f40526203a0>
```

Should we include `PresetFormat` here as an acceptable type too, or is it implied enough by `int` since it's an enum?

Reply:
I think, we should. At least as long as https://github.com/vapoursynth/vapoursynth/pull/572 is not merged.
Afterwards you don't need it at all anymore.